.pre-commit-config.yaml

```diff
@@ -0,0 +1,12 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
```

Comment:
https://pre-commit.ci/

you can add a `ci:` section and it will include running pre-commit in pr-checks without adding anything to the github workflows.  it'll also periodically update the `rev` number via PR.